sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystem.java

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.nio;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.util.CoreUtils;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.common.StorageSharedKeyCredential;
+import com.azure.storage.common.policy.RequestRetryOptions;
+import com.azure.storage.common.policy.RetryPolicyType;
+
+import java.io.IOException;
+import java.nio.file.FileStore;
+import java.nio.file.FileSystem;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.WatchService;
+import java.nio.file.attribute.UserPrincipalLookupService;
+import java.nio.file.spi.FileSystemProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * In the hierarchy of this file system, an {@code AzureFileSystem} corresponds to an Azure Blob Storage account. A
+ * file store is represented by a container in the storage account. Each container has one root directory.
+ *
+ * Closing the file system will not block on outstanding operations. Any operations in progress will be allowed to
+ * terminate naturally after the file system is closed, though no further operations may be started after the parent
+ * file system is closed.
+ * {@inheritDoc}
+ */
+public final class AzureFileSystem extends FileSystem {
+    // Configuration constants for blob clients.
+    public static final String AZURE_STORAGE_ACCOUNT_KEY = "AzureStorageAccountKey";
```

Comment:
Is there a reason you chose to put the constants in FileSystem and not FileSystemProvider?

Reply:
I put them here because they're used to configure the FileSystem, but since you use them to pass a map into a method on FileSystemProvider, I can see an argument for putting them on the provider.